_inc/client/setup-wizard/checklist-answer/index.jsx

```diff
@@ -0,0 +1,111 @@
+/**
+ * External dependencies
+ */
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import Gridicon from 'components/gridicon';
+
+import './style.scss';
+
+class ChecklistAnswer extends Component {
```

Comment:
As we move away from classes in other areas of Jetpack, should we do the same here as well? 

Related: #15694.

Reply:
I started with this as a functional component but I believe I couldn't get one of the code checker tools to accept it because the `set` function from `useState()` is generally called from an arrow function which isn't accepted by one of the code checkers. So I elected to switch to a class component and just put the functions on the class to avoid arrow functions.

I believe I could instead declare functions like `function foo() { ... }` inside a functional component, but that doesn't avoid the root problem of creating a function on each render so I decided not to. For each call to `setState` I would have to create a corresponding function like `function setVariableState( prop ) { setVariable( prop ); }` to avoid having to use an arrow function like `onChange={ ( prop ) => { setVariable( prop ) }` in the JSX.

All that said, I do prefer functional components to class components so I'd like to see us solve this!

Reply:
We discussed this in Slack but I wanted to put it here for anyone who may be following:

This issue is solved by the `useCallback` hook as shown in the current functional implementation in this PR.